src/test/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitorSpec.scala

```diff
@@ -46,28 +49,43 @@ class SubmissionMonitorSpec(_system: ActorSystem) extends TestKit(_system) with
   }
 
   it should "transition to running then completed then terminate" in withDefaultTestDatabase { dataSource =>
-    val monitorRef = TestActorRef[SubmissionMonitor](SubmissionMonitor.props(testData.wsName, testData.submission1, containerDAO, dataSource, 10 milliseconds, 1 second, TestActor.props()))
-    watch(monitorRef)
+	  val monitorRef = TestActorRef[SubmissionMonitor](SubmissionMonitor.props(testData.wsName, testData.submission1, containerDAO, dataSource, 10 milliseconds, 1 second, TestActor.props()))
+	  watch(monitorRef)
+
+	  //We're gonna store actor references here because actorSelection isn't super reliable.
+	  val wfActors = MMap[String, ActorRef]()
+
+	  //Give the submission monitor a chance to spawn the workflow actors first
+	  testData.submission1.workflows.foreach { workflow =>
+      awaitCond({
+        val tr = Try(Await.result(system.actorSelection(monitorRef.path / workflow.workflowId).resolveOne(100 milliseconds), Duration.Inf))
+        tr.foreach { actorRef =>
+          wfActors(workflow.workflowId) = actorRef
+        }
+        tr.isSuccess
+      }, 250 milliseconds)
+	  }
```

Comment:
More notes on this: `system.actorSelection` seems to not-always return an `ActorRef`, and doing a `tell` on an `ActorSelection` doesn't throw an error if the message doesn't go anywhere.

Instead, we explicitly try to resolve the `ActorSelection` to a real actor and keep a reference to it for the rest of the test. This block explicitly waits a bit (250ms) for the submission monitor to create its workflow actors; the `awaitCond` should barf if it's not created in that time, in which case we should investigate what's so hard about creating three actors in 250ms.

It's apparently good practise to send messages to `ActorRef`s rather than `ActorSelection`s, so we should keep this in mind.